src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -112,78 +112,129 @@ bool CImageLoaderTGA::isALoadableFileFormat(io::IReadFile* _file) const
 	return true;
 }
 
-// convertColorFlip() does color conversion as well as taking care of properly flipping the given image.
+/*
+	Targa formats needs two y-axis flips. The first is a flip to get the Y conforms to OpenGL coords.
+	The second flip is defined from within the .tga file itself (header.ImageDescriptor & 0x20).
+	if flip - perform two flips (OpenGL + Targa) = no flipping. Don't flip the image at all in that case
+	if not - do an OpenGL flip
+*/
+
 template <E_FORMAT srcFormat, E_FORMAT destFormat>
-static void convertColorFlip(uint32_t regionBufferRowLenght, VkExtent3D imageExtent, const core::smart_refctd_ptr<ICPUBuffer>& bufferSourceData, core::smart_refctd_ptr<ICPUBuffer>& bufferOutData, bool flip)
+core::smart_refctd_ptr<ICPUImage> createAndconvertImageData(ICPUImage::SCreationParams& imgInfo, core::smart_refctd_ptr<ICPUBuffer>& texelBuffer, core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>& regions, bool flip)
 {
-	const uint8_t *in = (const uint8_t*) bufferSourceData->getPointer();
-	uint8_t* out = (uint8_t*) bufferOutData->getPointer();
+	imgInfo.format = srcFormat;
+	using CONVERSION_FILTER = CConvertFormatImageFilter<srcFormat, destFormat>;
+	CONVERSION_FILTER convertFilter;
+	CONVERSION_FILTER::state_type state;
 
-	irr::core::vector3d size = 
+	core::smart_refctd_ptr<ICPUImage> inputCreationImage;
 	{
-		regionBufferRowLenght > 0 ? regionBufferRowLenght : imageExtent.width,
-		imageExtent.height,
-		imageExtent.depth
-	};
+		auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1u);
+		ICPUImage::SBufferCopy& region = regions->front();
+
+		region.imageSubresource.mipLevel = 0u;
+		region.imageSubresource.baseArrayLayer = 0u;
+		region.imageSubresource.layerCount = 1u;
+		region.bufferOffset = 0u;
+		region.bufferRowLength = asset::IImageAssetHandlerBase::calcPitchInBlocks(imgInfo.extent.width, asset::getTexelOrBlockBytesize(srcFormat));
+		region.bufferImageHeight = 0u;
+		region.imageOffset = { 0u, 0u, 0u };
+		region.imageExtent = imgInfo.extent;
+
+		inputCreationImage = asset::ICPUImage::create(std::move(imgInfo));
+		inputCreationImage->setBufferAndRegions(std::move(texelBuffer), regions);
+	}
```

Comment:
you should not be firing off a conversion if `srcFormat==destFormat`

Reply:
okay

Reply:
done